hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -111,15 +114,17 @@ object HoodieSparkSqlWriter {
             extraPreCommitFn: Option[BiConsumer[HoodieTableMetaClient, HoodieCommitMetadata]] = Option.empty):
   (Boolean, HOption[String], HOption[String], HOption[String], SparkRDDWriteClient[_], HoodieTableConfig) = {
 
-    assert(optParams.get("path").exists(!StringUtils.isNullOrEmpty(_)), "'path' must be set")
-    val path = optParams("path")
+    assert(rawParams.get("path").exists(!StringUtils.isNullOrEmpty(_)), "'path' must be set")
+    val path = rawParams("path")
     val basePath = new Path(path)
 
     val spark = sqlContext.sparkSession
     val sparkContext = sqlContext.sparkContext
 
     val fs = basePath.getFileSystem(sparkContext.hadoopConfiguration)
     tableExists = fs.exists(new Path(basePath, HoodieTableMetaClient.METAFOLDER_NAME))
+    var optParams = mayBeInferPartition(rawParams)
```

Comment:
minor: `optParams` sounds more intuitive indicating it comes from option(). you can call it `finalOptParams` after making inferences.